contribution_plan/signals.py

```diff
@@ -0,0 +1,22 @@
+import json
+
+from .models import ContributionPlan
+from core.signals import Signal
+from django.core.serializers.json import DjangoJSONEncoder
+from django.conf import settings
+from django.core.mail import send_mail, BadHeaderError
+
+_get_contribution_length_signal_params = ["contribution_plan"]
+get_contribution_length_signal = Signal(providing_args=_get_contribution_length_signal_params)
+
+
+def on_get_contribution_length_signal(sender, **kwargs):
```

Comment:
This will be in the calculation rule module not here because grace period will be (or not) defined in the calculation rule,